### DISCUSSION TITLE
[Enhancement] HarmonyOS

### DISCUSSION THREAD

**Discussion by jjchiw** [dotnet/maui#164] [General]
Hi!

I don't know how much is going to differ Base Android from Huawei Android (HarmonyOS), but it seems Huawei is creating their own apis for a lot of features :(

![image](https://user-images.githubusercontent.com/642319/86129262-b4634000-bae2-11ea-8591-634b4949f47e.png)

I just happen to have a fragmentation with the Camera2Api, It took me almost 2 weeks to realize that my camera app was not working in Huawei devices because they don't support Camera2Api, they created their own https://developer.huawei.com/consumer/en/CameraKit

It seems to be sort of easy to create the bindings (I'm going to start today to integrate it) 

- https://developer.huawei.com/consumer/en/doc/development/HMS-Guides/hms-location-xamarin-libbinding
- https://developer.huawei.com/consumer/en/doc/development/HMS-Library/hms-location-xamarin-sdkdownload

But since Xamarin Forms now supports Tizen maybe you can take a look at HarmonyOS

So It would be great if it could be out of the box, not sure if this enhancement should be in Xamarin Forms...

Thanks...

**Comment by mrwcjoughin** [comment]
Samsung currently maintains the Tizen implementation of Xamarin.Forms.
So my guess is Huawei would need to approach Microsoft and the community and do the same?

**Comment by dongfo** [comment]
dotnet Will most likely not support HarmonyOS, and Huawei will not have the willingness to actively adapt to dotnet in the short term. Thus, dotnet has a high probability of being excluded from the HarmonyOS ecology
